noir-projects/aztec-nr/aztec/src/test/mocks/mock_note.nr

```diff
@@ -76,6 +76,14 @@ impl NoteInterface<MOCK_NOTE_LENGTH, MOCK_NOTE_BYTES_LENGTH> for MockNote {
     }
 }
 
+impl Eq for MockNote {
+    fn eq(self, other: Self) -> bool {
+      (self.header.contract_address == other.header.contract_address) & 
+      (self.header.storage_slot == other.header.storage_slot) & 
```

Comment:
Agree on this, otherwise think it looks fine 👀

Reply:
I did it this way because `NoteHeader` includes extra fields (`nonce` and `is_transient`), which `note_getter` does not constrain - I dion't really understand those so I'm not sure why they would not be checked.